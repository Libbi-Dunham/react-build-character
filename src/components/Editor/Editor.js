import React from 'react';

export default function Editor({
  catchphrase,
  setHeadCount,
  setHead,
  setMiddle,
  setBottom,
  setCatchphrase,
  setCatchphraseList,
  setCatchphraseCount,
  // handleHeadCount,
  handleMiddleCount,
  handleBottomCount,
  // handleClick,
  // handleCatchphraseCount,
}) {
  const handleClick = () => {
    setCatchphraseList((prevState) => [...prevState, catchphrase]);

    setCatchphrase('');
  };

  const handleCatchphraseCount = (e) => {
    setCatchphrase(e.target.value);
    setCatchphraseCount((prevState) => prevState + 1);
  };

  const handleHeadCount = (e) => {
    setHead(e.target.value);
    setHeadCount((prevState) => prevState + 1);
  };

  return (
    <>
      <div className="editor">
        <label>Head</label>
        <select value={setHead} onChange={handleHeadCount}>
          <option value="Bird">Bird</option>
          <option value="Duck">Duck</option>
          <option value="Dog">Dog</option>
          <option value="Horse">Horse</option>
        </select>
      </div>

      <div className="editor">
        <label>Middle</label>
        <select value={setMiddle} onChange={handleMiddleCount}>
          <option value="Blue">Blue</option>
          <option value="Fancy">Fancy</option>
          <option value="Pink">Pink</option>
          <option value="Red">Red</option>
        </select>
      </div>

      <div className="editor">
        <label>Bottom</label>
        <select value={setBottom} onChange={handleBottomCount}>
          <option value="leg">Single Leg</option>
          <option value="white">White Pants</option>
          <option value="blue">Blue Jeans</option>
          <option value="dog">Dog Pants</option>
        </select>
      </div>

      <div className="editor">
        <label>Catch Phrase!</label>
        <input type="text" onChange={(e) => setCatchphrase(e.target.value)} />
        <button onChange={handleClick}>Add!</button>
      </div>
    </>
  );
}
