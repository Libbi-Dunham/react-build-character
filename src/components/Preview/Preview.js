import React from 'react';

import './Preview.css';

export default function Preview({
  head,
  middle,
  bottom,
  catchphrase,
  // setHeadCount,
  // setMiddleCount,
  // setBottomCount,
  // setHead,
  // setMiddle,
  // setBottom,
}) {
  return (
    <>
      <div className={`preview ${catchphrase}`}>
        <img height="75" alt={head} src={`${process.env.PUBLIC_URL}/${head}-head.png`} />
        <img height="75" alt={middle} src={`${process.env.PUBLIC_URL}/${middle}-middle.png`} />
        <img height="75" alt={bottom} src={`${process.env.PUBLIC_URL}/${bottom}-pants.png`} />
        <p> {catchphrase} </p>
      </div>
      {/* <div>
        <p>You changed the head {setHeadCount} times</p>
        <p>You changed the middle {setMiddleCount} times</p>
        <p>You changed the Bottom {setBottomCount} times</p>
      </div> */}
    </>
  );
}
