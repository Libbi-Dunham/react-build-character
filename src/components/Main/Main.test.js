import { render } from '@testing-library/react';
import Preview from '../Preview/Preview';
import Editor from '../Editor/Editor';

test('both Editor and Preview to pop up', () => {
  const container = render(
    <>
      <Preview />
      <Editor />
    </>
  );
  expect(container).toMatchSnapshot();
});
