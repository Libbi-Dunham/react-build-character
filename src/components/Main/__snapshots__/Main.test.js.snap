// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`both Editor and Preview to pop up 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="clothes"
      >
        <img
          height="75"
          src="/undefined-head.png"
        />
        <img
          height="75"
          src="/undefined-middle.png"
        />
        <img
          height="75"
          src="/undefined-pants.png"
        />
        <p>
           
           
        </p>
      </div>
      <div
        class="editor"
      >
        <label>
          Head
        </label>
        <select>
          <option
            value="Bird"
          >
            Bird
          </option>
          <option
            value="Duck"
          >
            Duck
          </option>
          <option
            value="Dog"
          >
            Dog
          </option>
          <option
            value="Horse"
          >
            Horse
          </option>
        </select>
      </div>
      <div
        class="editor"
      >
        <label>
          Middle
        </label>
        <select>
          <option
            value="Blue"
          >
            Blue
          </option>
          <option
            value="dress"
          >
            Fancy
          </option>
          <option
            value="Pink"
          >
            Pink
          </option>
          <option
            value="Red"
          >
            Red
          </option>
        </select>
      </div>
      <div
        class="editor"
      >
        <label>
          Bottom
        </label>
        <select>
          <option
            value="leg"
          >
            Single Leg
          </option>
          <option
            value="white"
          >
            White Pants
          </option>
          <option
            value="blue"
          >
            Blue Jeans
          </option>
          <option
            value="dog"
          >
            Dog Pants
          </option>
        </select>
      </div>
      <div
        class="editor"
      >
        <label>
          Catch Phrase!
        </label>
        <input
          type="text"
          value=""
        />
        <button>
          Add!
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="clothes"
    >
      <img
        height="75"
        src="/undefined-head.png"
      />
      <img
        height="75"
        src="/undefined-middle.png"
      />
      <img
        height="75"
        src="/undefined-pants.png"
      />
      <p>
         
         
      </p>
    </div>
    <div
      class="editor"
    >
      <label>
        Head
      </label>
      <select>
        <option
          value="Bird"
        >
          Bird
        </option>
        <option
          value="Duck"
        >
          Duck
        </option>
        <option
          value="Dog"
        >
          Dog
        </option>
        <option
          value="Horse"
        >
          Horse
        </option>
      </select>
    </div>
    <div
      class="editor"
    >
      <label>
        Middle
      </label>
      <select>
        <option
          value="Blue"
        >
          Blue
        </option>
        <option
          value="dress"
        >
          Fancy
        </option>
        <option
          value="Pink"
        >
          Pink
        </option>
        <option
          value="Red"
        >
          Red
        </option>
      </select>
    </div>
    <div
      class="editor"
    >
      <label>
        Bottom
      </label>
      <select>
        <option
          value="leg"
        >
          Single Leg
        </option>
        <option
          value="white"
        >
          White Pants
        </option>
        <option
          value="blue"
        >
          Blue Jeans
        </option>
        <option
          value="dog"
        >
          Dog Pants
        </option>
      </select>
    </div>
    <div
      class="editor"
    >
      <label>
        Catch Phrase!
      </label>
      <input
        type="text"
        value=""
      />
      <button>
        Add!
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
